common/events/events_io.cpp

```diff
@@ -123,8 +123,8 @@ void AsyncWriterFromWriter::Cancel() {
 	}
 }
 
-ReaderFromAsyncReader::ReaderFromAsyncReader(AsyncReaderFromEventLoopFunc func) {
-	reader_ = func(event_loop_);
+ReaderFromAsyncReader::ReaderFromAsyncReader(AsyncReaderFromEventLoopFunc func) :
+	reader_(func(event_loop_)) {
```

Comment:
This is actually potentially quite a big change, right?

Reply:
Isn't it doing the same thing?

Reply:
Constructing a default object and then assigning to it vs. copy-constructing an object? I mean, in this case it's (probably?) the same.

Reply:
It really should be, otherwise we are creating problems for ourselves.